bindings/pyroot_experimental/pythonizations/python/ROOT/pythonization/_tfile.py

```diff
@@ -15,12 +15,74 @@
 
 from libROOTPythonizations import AddFileOpenPyz
 from ROOT import pythonization
+import sys, errno, os
+from functools import partial
+from cppyy.gbl import gSystem
+
+# We check the Python version because FileNotFoundError
+# is not available in Python2
+if sys.version_info[0] > 2:
+    CustomFileNotFoundError = FileNotFoundError
+else:
+    CustomFileNotFoundError = IOError
+
+def _TFileConstructor(self, *args):
+    # Redefinition of ROOT.TFile(str, ...):
+    # checks if a file with the given name exists and raises
+    # a FileNotFoundError-like if not.
+    # Parameters:
+    # self: instance of TFile class
+    # *args: arguments passed to the constructor; only args[0]
+    # (string containing the name of the file) is mandatory
+
+    # Check the existance of the file (and fail) only in READ
+    # mode
+    if len(args) == 1 or (len(args) == 2 and args[1] == 'READ'):
+        file_name = args[0]
+        if isinstance(file_name, str) and gSystem.AccessPathName(file_name):
+            raise CustomFileNotFoundError(errno.ENOENT,
+                    os.strerror(errno.ENOENT),
+                    file_name)
+    self._OriginalConstructor(*args)
+    if self is None or self.IsZombie():
+        raise CustomFileNotFoundError(errno.ENOENT,
+                os.strerror(errno.ENOENT),
+                file_name)
+
+def _TFileOpen(klass, *args):
+    # Redefinition of ROOT.TFile.Open(str, ...):
+    # checks if a file with the given name exists and raises
+    # a FileNotFoundError-like if not.
+    # Parameters:
+    # klass: TFile class
+    # *args: arguments passed to the constructor; only args[0]
+    # (string containing the name of the file) is mandatory
+
+    # Check the existance of the file (and fail) only in READ
+    # mode
+    if len(args) == 1 or (len(args) == 2 and args[1] == 'READ'):
```

Comment:
This check of the arguments might not be even be necessary, I would just check the return value of the original Open and raise and exception if the internal object is null. And similarly for the constructor.